library/timing.c

```diff
@@ -266,18 +267,16 @@ unsigned long mbedtls_timing_get_timer( struct mbedtls_timing_hr_time *val, int
 /* It's OK to use a global because alarm() is supposed to be global anyway */
 static DWORD alarmMs;
 
-static DWORD WINAPI TimerProc( LPVOID TimerContext )
+static void TimerProc( void *TimerContext )
 {
-    ((void) TimerContext);
+    (void)TimerContext;
     Sleep( alarmMs );
     mbedtls_timing_alarmed = 1;
-    return( TRUE );
+    // Implicit call of _endthread() is better (see MS online docs)
```

Comment:
What do you mean by this comment? _endthread() is called automatically upon returning, but how is it better and compared to what?

Reply:
Better than explicit call.
The reason is that `_endthread()` does not return, and all following code would not be executed.
It could be "invisible" cleanup code, added by compiler. Then memory leaks or incorrect exception unwinding would be possible.

Reply:
What I meant by my comment was - it's better to update the comment with exact reasons, instead of sending people to vague docs, without specific paragraphs :)
So here, it's good to add:
"explicit _endthread call causes C++ destructors pending in the thread not to be called.", or a summary of what you've written above.

Reply:
Please see if the updated comment is adequate.